File: src/core/services/connection.service.ts
import { invoke } from '@tauri-apps/api/core';
import type { ConnectionConfig, SshConnectionInfo } from '../../proto';

export async function listConnections(): Promise<ConnectionConfig[]> {
  return invoke('list_connections');
}

export async function saveConnection(config: ConnectionConfig): Promise<void> {
  return invoke('save_connection', { config });
}

export async function deleteConnection(id: string): Promise<void> {
  return invoke('delete_connection', { id });
}

export async function testConnection(ssh: SshConnectionInfo): Promise<string> {
  return invoke('test_connection', { ssh });
}

File: src-tauri/src/app/connection_service.rs
use crate::core::error::Result;
use crate::core::types::ConnectionConfig;
use crate::infra::storage::database::Database;
use crate::infra::storage::connection_repo::ConnectionRepo;

pub struct ConnectionService;

impl ConnectionService {
    pub fn list_connections(db: &Database) -> Result<Vec<ConnectionConfig>> {
        ConnectionRepo::list(db)
    }

    pub fn save_connection(db: &Database, config: &ConnectionConfig) -> Result<()> {
        ConnectionRepo::save(db, config)
    }

    pub fn delete_connection(db: &Database, id: &str) -> Result<()> {
        ConnectionRepo::delete(db, id)
    }
}
